Add http://freshon.tv torrent provider.

damoxc opened the pull request:
This adds the abilility to fetch torrents from http://freshon.tv. It checks
their RSS feed using a users passkey to fetch the feed with no login
required.

damoxc committed:
providers: add freshon.tv support

This adds the abilility to fetch torrents from http://freshon.tv. It checks
their RSS feed using a users passkey to fetch the feed with no login
required.

jness commented:
I've been testing your commit but it seems the standard SickBeard search pattern is not sufficient for Freshon.tv.

The pattern SB is using is 'Weeds - 7x02 - From Trauma Cometh Something':

```
     Unable to find a download for Weeds - 7x02 - From Trauma Cometh Something
```

This pattern is based on EZRSS's naming convention, however Freshon.tv seems to use a 'Weeds.S07E02' Approach.

Any chances for a new commit to conform to this behaviour?

damoxc commented:
I'm not sure if that is the issue actually. I've been the provider for a number of weeks now and it successfully matched:

Weeds.S07E02.720p.HDTV.X264-DIMENSION

As well as everything else from Freshon I'm subscribed to. Do you have anything else in the SB log that can give a clue as to what it is?

jness commented:
Hmm... I'm wondering if this has something to do with me pulling your fork down and copying in my cache_, sickbeard.db_, and config.ini from my working branch..

```
           $ tail -f sickbeard.log  -n0
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: 74845: An object for episode 7x2 didn't exist in the cache, trying to create it
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: 74845: Loading episode details from DB for episode 7x2
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: sickbeard.db: SELECT * FROM tv_episodes WHERE showid = ? AND season = ? AND episode = ? with args [74845, 7, 2]
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: 74845: Loading episode details from theTVDB for episode 7x2
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: 74845: Setting status for 7x2 based on status 104 and existence of /home/jness/TV Shows/Weeds/Weeds - 7x02 - From Trauma Cometh Something.avi
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: Checking if /home/jness/TV Shows/Weeds/Weeds - 7x02 - From Trauma Cometh Something.nfo exists: True
           Jul-08 12:32:55 DEBUG    CP Server Thread-2 :: Checking if /home/jness/TV Shows/Weeds/Weeds - 7x02 - From Trauma Cometh Something.tbn exists: True
           Jul-08 12:32:58 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for download for Weeds - 7x02 - From Trauma Cometh Something
           Jul-08 12:32:58 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for Weeds - 7x02 - From Trauma Cometh Something
           Jul-08 12:32:58 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching EZRSS for Weeds - 7x02 - From Trauma Cometh Something
           Jul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['ezrss']
           Jul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['ezrss']
           Jul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Last update was too soon, using old cache: today()-2011-07-08 12:31:37<0:15:00
           Jul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT * FROM ezrss WHERE tvdbid = ? AND season = ? AND episodeJul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT * FROM ezrss WHERE tvdbid = ? AND season = ? AND episodeJul-08 12:32:QUEUE-MANUAL-SEARCH :: Search string: https://www.ezrss.it/search/index.php?season=7&show_name=Weeds&episode=2&mode=rss
           Jul-08 12:32:58 ERROR    SEARCHQUEUE-MANUAL-SEARCH :: Error loading EZRSS URL: (<class 'urllib2.HTTPError'>, HTTPError(), <traceback object at 0x928faa4>) - 
           Jul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Picking the best result out of []
           Jul-08 12:32:58 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: No result picked.
           Jul-08 12:32:58 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Unable to find a download for Weeds - 7x02 - From Trauma Cometh Something
```

I'm going to try a fresh pull and fresh config

jness commented:
Fresh install produced the same issue:

```
Jul-08 12:44:45 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for download for Weeds - 7x02 - From Trauma Cometh Something
Jul-08 12:44:45 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for Weeds - 7x02 - From Trauma Cometh Something
Jul-08 12:44:45 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching FreshOnTv for Weeds - 7x02 - From Trauma Cometh Something
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['freshontv']
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['freshontv']
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Last update was too soon, using old cache: today()-2011-07-08 12:41:11<0:15:00
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT * FROM freshontv WHERE tvdbid = ? AND season = ? AND episodes LIKE ? with args [74845, 7, '|2|']
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Cache results: []
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Picking the best result out of []
Jul-08 12:44:45 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: No result picked.
Jul-08 12:44:45 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Unable to find a download for Weeds - 7x02 - From Trauma Cometh Something
```

damoxc commented:
Just going to check that you have put your passkey in on the configuration page?

damoxc committed:
providers: check for passkey in freshon.tv provider

Currently the provider does no checking to make sure that the user
has entered a passkey. If we don't have a passkey then the rss
feed redirects us to a login page causing all searching to fail.

jness commented:
Yes I verified my passkey is in the config.ini and enabled is set to '1'. 

However, in the interface it shows the provider disabled and no passkey set (even though it is in the config.ini).

Thoughts?

damoxc commented:
It could be the config file is not being loaded correctly by my provider. Could you verify it works okay when you enable it and enter the passkey on the configuration page. And I will check the config loading.

jness commented:
I've updated the Provider via the interface and verify it displays enabled, however it is still not finding anything:

```
Jul-08 13:03:18 DEBUG    CP Server Thread-5 :: 74845: An object for episode 7x2 didn't exist in the cache, trying to create it
Jul-08 13:03:18 DEBUG    CP Server Thread-5 :: 74845: Loading episode details from DB for episode 7x2
Jul-08 13:03:18 DEBUG    CP Server Thread-5 :: sickbeard.db: SELECT * FROM tv_episodes WHERE showid = ? AND season = ? AND episode = ? with args [74845, 7, 2]
Jul-08 13:03:18 DEBUG    CP Server Thread-5 :: 74845: Loading episode details from theTVDB for episode 7x2
Jul-08 13:03:18 DEBUG    CP Server Thread-5 :: 74845: Setting status for 7x2 based on status 5 and existence of 
Jul-08 13:03:18 DEBUG    CP Server Thread-5 :: Not touching status because we have no ep file, the airdate is in the past, and the status is 5
Jul-08 13:03:19 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for download for Weeds - 7x02 - From Trauma Cometh Something
Jul-08 13:03:19 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for Weeds - 7x02 - From Trauma Cometh Something
Jul-08 13:03:19 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching FreshOnTv for Weeds - 7x02 - From Trauma Cometh Something
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['freshontv']
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['freshontv']
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Last update was too soon, using old cache: today()-2011-07-08 13:01:48<0:15:00
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT * FROM freshontv WHERE tvdbid = ? AND season = ? AND episodes LIKE ? with args [74845, 7, '|2|']
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Cache results: []
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Picking the best result out of []
Jul-08 13:03:19 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: No result picked.
Jul-08 13:03:19 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Unable to find a download for Weeds - 7x02 - From Trauma Cometh Something
```

damoxc committed:
providers: fix loading freshon.tv config

Missing globals in the __init__ file

damoxc commented:
Okay my latest commits fix loading the configuration. Could you grep your logs for 'FreshOnTV cache update URL' to see what url it is requesting?

jness commented:
Ok, I pulled the latest sources from your Fork and started fresh, unfortunately It is still not pulling:

```
 Jul-08 14:37:47 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for download for Weeds - 7x02 - From Trauma Cometh Something
 Jul-08 14:37:47 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching for Weeds - 7x02 - From Trauma Cometh Something
 Jul-08 14:37:47 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Searching FreshOnTv for Weeds - 7x02 - From Trauma Cometh Something
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['freshontv']
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT time FROM lastUpdate WHERE provider = ? with args ['freshontv']
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Last update was too soon, using old cache: today()-2011-07-08 14:36:57<0:15:00
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: cache.db: SELECT * FROM freshontv WHERE tvdbid = ? AND season = ? AND episodes LIKE ? with args [74845, 7, '|2|']
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Cache results: []
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: Picking the best result out of []
 Jul-08 14:37:47 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: No result picked.
 Jul-08 14:37:47 INFO     SEARCHQUEUE-MANUAL-SEARCH :: Unable to find a download for Weeds - 7x02 - From Trauma Cometh Something
```

jness commented:
Here is the 'grep':

```
 $ grep "FreshOnTV cache update URL" Logs/sickbeard.log 
 Jul-08 14:36:56 DEBUG    SEARCHQUEUE-MANUAL-SEARCH :: FreshOnTV cache update URL: http://freshon.tv/rss.php?feed=dl&passkey=XXXXXXXXXXXXXXXXXXXX
```

damoxc commented:
If you go to the URL directly do you see a RSS feed?

jness commented:
Yes when I access that URL I am provided with the XML feed:

Snippet below:

```
 <channel>
 <title>Tv TORRENTs</title>
 <description>TvT RSS FEED</description>
 <link/>
 <lastBuildDate>Fri, 08 Jul 2011 23:10:37 GMT</lastBuildDate>
 <copyright>Copyright Tv TORRENTs</copyright>
 <managingEditor>admin@freshon.tv</managingEditor>
 <webMaster>admin@freshon.tv</webMaster>
 <ttl>5</ttl>
 <image>
 <title>Tv TORRENTs : Content</title>
 <url>http://freshon.tv/fav.ico</url>
 <link/>
 <width>16</width>
 <height>16</height>
 <description>TV Shows</description>
 </image>
 <item>
 <title>Fifth.Gear.S19E09.WS.PDTV.XviD-C4TV</title>
 <category>Fifth Gear</category>
 <pubDate>Fri, 08 Jul 2011 22:49:03 +0300</pubDate>
 <link>
 http://freshon.tv/download.php?type=rss&id=70857&passkey=XXXXXXXXXXXX
 </link>
```

jness commented:
This RSS Feed is not providing the show in question, I guess it only shows the latest X uploads??

damoxc commented:
Yes unfortunately it only shows up to day normally. I imagine you
could get around it by supplying categories in the url but that gets
complicated then. Would have to create some way to get the category id
from the site and then provide those for all the required shows.

jness commented:
Ahh ok. I'll keep that in mind and see how it works out.

jness commented:
Everything is working great (as long as the torrent is on the latest RSS feed).

Thanks!

damoxc commented:
Excellent news, thanks for testing :-)

damoxc committed:
Merge remote-tracking branch 'midgetspy/master'

Conflicts:
	sickbeard/webserve.py

damoxc committed:
Merge remote-tracking branch 'upstream/master'

Conflicts:
	data/interfaces/default/config_providers.tmpl
	sickbeard/__init__.py
	sickbeard/providers/__init__.py
	sickbeard/webserve.py

thezoggy commented:
can you rebase this off the development branch and updated the code where needed? there has been many changes since this was submitted...

thezoggy commented:
freshon.tv points to tv romanian tracker... not sure if content is in english?
